editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -2525,6 +2525,12 @@ bool CanvasItemEditor::_gui_input_ruler_tool(const Ref<InputEvent> &p_event) {
 		ruler_tool_origin = snap_point(viewport->get_local_mouse_position() / zoom + view_offset);
 	}
 
+	if (b.is_valid() && b->get_button_index() == MouseButton::RIGHT && ruler_tool_active) {
```

Comment:
```suggestion
	if (ruler_tool_active && b.is_valid() && b->get_button_index() == MouseButton::RIGHT) {
```
Cheaper checks first.